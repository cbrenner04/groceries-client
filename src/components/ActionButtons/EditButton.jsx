import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'react-bootstrap';

const EditButton = ({ handleClick, testID, disabled }) => (
  <Button variant="link" onClick={handleClick} className="p-0 mr-3" disabled={disabled} data-test-id={testID}>
    <i className="fa fa-edit fa-2x text-warning" />
  </Button>
);

EditButton.propTypes = {
  handleClick: PropTypes.func.isRequired,
  testID: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
};

EditButton.defaultProps = {
  disabled: false,
};

export default EditButton;
