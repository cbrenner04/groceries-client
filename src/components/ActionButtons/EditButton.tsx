import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'react-bootstrap';

const EditButton = ({ handleClick, testID, disabled }) => (
  <Button variant="link" onClick={handleClick} className="p-0 me-3" disabled={disabled || false} data-test-id={testID}>
    <i className="fa fa-edit fa-2x text-warning" />
  </Button>
);

EditButton.propTypes = {
  handleClick: PropTypes.func.isRequired,
  testID: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
};

export default EditButton;
