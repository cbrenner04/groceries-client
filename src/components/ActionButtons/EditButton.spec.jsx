import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import EditButton from './EditButton';

describe('EditButton', () => {
  let props;

  beforeEach(() => {
    props = {
      handleClick: jest.fn(),
      'data-test-id': 'test-id',
    };
  });

  it('renders', () => {
    const { container, getByTestId } = render(<EditButton {...props} />);

    expect(container).toMatchSnapshot();
    expect(getByTestId('test-id')).toBeVisible();
  });

  it('calls handleClick on click', () => {
    const { getByTestId } = render(<EditButton {...props} />);

    fireEvent.click(getByTestId('test-id'));

    expect(props.handleClick).toHaveBeenCalled();
  });
});
