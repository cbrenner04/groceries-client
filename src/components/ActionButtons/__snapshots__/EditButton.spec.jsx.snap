// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditButton renders 1`] = `
<div>
  <button
    class="p-0 mr-3 btn btn-link"
    data-test-id="test-id"
    type="button"
  >
    <i
      class="fa fa-edit fa-2x text-warning"
    />
  </button>
</div>
`;
